hamocc/mo_parambgc_ini.F90

```diff
@@ -0,0 +1,686 @@
+                          & pi_alpha,rcalc,rcar, rdn2o1,rdn2o2,rdnit0,rdnit1,rdnit2,relaxfe,remido,riron,rnit,rnoi,ro2ut,          &
+                          & ropal,spemor,tf0,tf1,tf2,tff,wcal,wdust,wopal,wpoc,zinges 
+  use mo_sedmnt,      only: claydens,o2ut,rno3
+  use mo_control_bgc, only: io_stdo_bgc,bgc_namelist
+  use mo_param1_bgc,  only: iatmco2,iatmnco2,iatmo2,iatmn2,iatmc13,iatmc14,iatmbromo
+  use mod_xc,         only: mnproc
+
+#ifdef AGG
+  use mo_biomod,      only: alar1,alar2,alar3,alow1,alow2,alow3,calmax,cellmass,cellsink,dustd1,dustd2,dustd3,dustsink,            &
+                          & fractdim,fse,fsh,nmldmin,plower,pupper,safe,sinkexp,stick,tmfac,tsfac,vsmall,zdis
+#elif defined(WLIN)
+  use mo_biomod,      only: wmin,wmax,wlin
+#endif
+#ifdef BROMO
+  use mo_biomod,      only: rbro
+  use mo_carbch,      only: atm_bromo,fbro1,fbro2
+#endif
+#ifdef cisonew
+  use mo_biomod,      only: bifr13,bifr14,c14fac,prei13,prei14,re1312,re14to
+  use mo_carbch,      only: atm_c13, atm_c14,c14_t_half,c14dec
+#endif
+#ifdef natDIC
+  use mo_carbch,      only: atm_co2_nat
+#endif
+
+  implicit none
+
+  private
+   
+  public :: ini_parambgc
+
+  ! Module-wide parameters (used in more than one subroutine)
+#ifndef AGG
+  REAL :: dustd1, dustd2, dustsink
+#endif
+#ifdef cisonew 
+  REAL :: beta13, alpha14, d14cat, d13c_atm  
+#endif
+
+  contains
+
+  !---------------------------------------------------------------------------------------------------------------------------------
+  subroutine ini_parambgc(kpie,kpje)
+    !
+    ! First, Initialze parameters of individual components with default values. 
+    ! The order of initialization can matter due to interdependcies.
+    ! Then read the namelist and adjust rates to 'per time step'
+    ! Re-adjust dependent parameter values
+    ! Eventually write out the used parameters to the log file
+    !
+    implicit none      
+
+    INTEGER, intent(in) :: kpie,kpje
+
+    call ini_param_atm()           ! Initialize default atmospheric parameters
+    call ini_stoichiometry()       ! Initialize fixed stoichiometric parameters 
+    call ini_param_biol()          ! initialize biological parameters
+#ifdef AGG
+    call ini_aggregation()         ! Initialize aggregation module of Iris Kriest (no NML read thus far)
+#endif
+
+    call read_bgcnamelist()        ! read the BGCPARAMS namelist
+    call calc_param_atm()          ! calculate atmospheric parameters after updating parameters via nml
+    call ini_fields_atm(kpie,kpje) ! initialize atmospheric fields with (updated) parameter values 
+    call readjust_param()          ! potentially readjust namlist parameter-dependent parameters
+    call rates_2_timestep()        ! Converting rates from /d... to /dtb
+
+    call write_parambgc()          ! write out used parameters and calculate back rates from /dtb to /d..
```

Comment:
Wouldn't it make more sense to write out the parameter BEFORE multiplying by `dtb`? If I want to check the settings of parameters in the log file, I relate much more to the value that I have set via namelist than the value multiplied by `dtb`.